packages/sync/src/modules/Full_Sync.php

```diff
@@ -700,4 +771,57 @@ private function read_option( $name, $default = null ) {
 
 		return $value;
 	}
+
+	/**
+	 * Prefix of the blog lock transient.
+	 *
+	 * @access public
+	 *
+	 * @var string
+	 */
+	const ENQUEUE_LOCK_TRANSIENT_PREFIX = 'jp_sync_enqueue_lock_';
+	/**
+	 * Lifetime of the blog lock transient.
+	 *
+	 * @access public
+	 *
+	 * @var int
+	 */
+	const ENQUEUE_LOCK_TRANSIENT_EXPIRY = 15; // Seconds.
+
+	/**
+	 * Attempt to lock enqueueing when the server receives concurrent requests from the same blog.
+	 *
+	 * @access public
+	 *
+	 * @param int $expiry  enqueue transient lifetime.
+	 * @return boolean True if succeeded, false otherwise.
+	 */
+	public function attempt_enqueue_lock( $expiry = self::ENQUEUE_LOCK_TRANSIENT_EXPIRY ) {
```

Comment:
In multi-data-center environments, transients might be implemented via local memcached and so this technique might not be reliable.

Reply:
should we use an option?